app/models/maintenance_tasks/task.rb

```diff
@@ -306,12 +359,35 @@ def cursor_columns
     # Placeholder method to raise in case a subclass fails to implement the
     # expected instance method.
     #
-    # @param _item [Object] the current item from the enumerator being iterated.
+    # When the Task is parallelized, this method processes a batch by spawning
+    # threads for parallel execution. Otherwise, it raises an error advising
+    # subclasses to implement an override.
     #
-    # @raise [NotImplementedError] with a message advising subclasses to
+    # @param item_or_batch [Object] the current item from the enumerator being
+    #   iterated, or a batch when parallelized.
+    #
+    # @raise [NoMethodError] with a message advising subclasses to
+    #   implement an override for this method.
+    def process(item_or_batch)
+      if self.class.parallelized?
+        process_batch_in_parallel(item_or_batch)
+      else
+        raise NoMethodError, "#{self.class.name} must implement `process`."
+      end
+    end
+
+    # Task authors implement this method instead of process(item) when using
+    # parallelize. It will be called in parallel for each item in a batch.
+    #
+    # @param _item [Object] the individual item to process
+    #
+    # @raise [NoMethodError] with a message advising subclasses to
     #   implement an override for this method.
-    def process(_item)
-      raise NoMethodError, "#{self.class.name} must implement `process`."
+    def process_item(_item)
```

Comment:
I'm not sure exactly what to name this, but I think we need an API that's more distinct from `#process` that indicates that this is for parallel processing in a batch. Maybe `#process_for_batch`?